### DISCUSSION TITLE
How does this library stream RTP over websocket when ws runs on TCP and not UDP?

### DISCUSSION THREAD

**Discussion by Erikmmkarlsson** [AxisCommunications/media-stream-library-js#1047] [Q&A]
How does it work that it uses websocket to stream the RTP to video in the browser, when websocket is TCP based, and RTP is based on UDP? I can't find in the code how that happens.

**Comment by steabert** [comment]
RTP doesn't need to be sent over UDP, it just usually is.

**Reply by Erikmmkarlsson** [reply]
I see now that this library relies on RTP over TCP, because it sends in the setup `Headers: {  Transport: RTP/AVP/TCP;` in rtsp-session component.
